app/packs/src/apps/mydb/elements/details/ViewSpectra.js

```diff
@@ -218,6 +218,9 @@ class ViewSpectra extends React.Component {
       peaks, layout, decimal, shift, isAscend, isIntensity, boundary, integration
     });
     const layoutOpsObj = SpectraOps[layout];
+    if (!layoutOpsObj) {
+      return [];
```

Comment:
Could this be an early return? Meaning, could the instantiation of `layoutObsObj` and the subsequent `if` statement be the first thing that's done in `formatPks`? Otherwise, in case there isn't a `layoutObsObj`, all the work that `formatPks` does prior to the instantiation of `layoutObsObj` is wasted.

Reply:
@JanCBrammer Updated, please have a look.

Reply:
Looks good.